### PR TITLE
Bugfix/sig 1420 pdf downloadknop werkt niet vanuit ie

### PR DESCRIPTION
**Pull request by jasperswart**

### CODE REVIEW

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js [hunk 1]
```diff
@@ -18,9 +18,13 @@ function handleDownload(url, filename, accessToken) {
     responseType: 'blob'
   }).then((response) => response.blob())
     .then((blob) => {
-      const link = document.querySelector('.download-button__link');
-      link.href = window.URL.createObjectURL(blob);
-      link.click();
+      if (navigator.msSaveOrOpenBlob) {
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: Don't forget to write a test for this.
See https://github.com/Amsterdam/registraties/blob/9e56932ad271dfb29fc574edd25476238c5e5c7f/app/containers/CSVDownload/__tests__/CSVDownload.test.js#L235 for reference.

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js [hunk 2]
```diff
@@ -5,39 +5,71 @@ import 'whatwg-fetch';
...
+      method: 'GET',
+      headers,
+      responseType: 'blob'
+    }).then((response) => response.blob())
+        .then((blob) => {
+          if (navigator.msSaveOrOpenBlob) {
+            navigator.msSaveOrOpenBlob(blob, filename);
+          } else {
+            const link = this.downloadLink.current;
+            link.href = window.URL.createObjectURL(blob);
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: See the [section on memory management](https://developer.mozilla.org/en-US/docs/Web/API/URL/createObjectURL#Memory_management) in the URL docs on MDN; you should revoke the object URL.
See example here: https://developer.mozilla.org/en-US/docs/Web/API/File/Using_files_from_web_applications#Example_Using_object_URLs_to_display_images

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js [hunk 3]
```diff
@@ -5,39 +5,71 @@ import 'whatwg-fetch';
...
+  render() {
+    const { label, url, filename, accessToken } = this.props;
+    return (
+      <div className="download-button align-self-center">
+        <button
+          className="incident-detail__button"
+          onClick={() => this.handleDownload(url, filename, accessToken)}
+        >{label}</button>
+
+        <a
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: Why both a `<button>` and an `<a>` element? Should be safe to remove the `<button>` and handle the click on the `<a>`.

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js [hunk 4]
```diff
@@ -5,39 +5,71 @@ import 'whatwg-fetch';
...
+          ref={this.downloadLink}
+          className="download-button__link"
+          download={filename}
+        ></a>
+      </div>
+    );
+  }
+
+}
+// const DownloadButton = ({ label, url, filename, accessToken }) => (
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: Remove this block of code.

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.test.js [hunk 5]
```diff
@@ -1,10 +1,68 @@
...
+import { shallow } from 'enzyme';
 
-// import DownloadButton from 'index';
+import DownloadButton from './index';
 
 describe('<DownloadButton />', () => {
-  it('Expect to have unit tests specified', () => {
-    expect(true).toEqual(true);
+  let props;
+  // let input;
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.test.js]: Remove commented out blocks of code.

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.test.js [hunk 6]
```diff
@@ -1,10 +1,68 @@
...
+    it('should render correctly when logged in', () => {
+      props.accessToken = 'access-token';
+      const wrapper = shallow(
+        <DownloadButton {...props} />
+      );
+      expect(wrapper).toMatchSnapshot();
+    });
   });
+
+  // later
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.test.js]: Remove

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js (hunk 2 above)
Comment (jasperswart) [reply src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: ik heb nu een cleanere opzet en de revoke erin

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js (hunk 4 above)
Comment (jasperswart) [reply src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: gedaan

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js (hunk 3 above)
Comment (jasperswart) [reply src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: alleen de anchor werkte niet, maar alleen de button wel: deze test op Chrome, Safari, FF en op Win IE11, en edge 15 en FF

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js [hunk 7]
```diff
@@ -5,39 +5,55 @@ import 'whatwg-fetch';
...
+          }
+        });
+  }
+
+  render() {
+    const { label, url, filename, accessToken } = this.props;
+    return (
+      <div className="download-button align-self-center">
+        <button
+          className="incident-detail__button"
```
Comment (janjaap) [inline src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: `<button>` Element has to have a `type` attribute with value 'button'

Comment (jasperswart) [reply src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: gedaan

File: src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js (hunk 1 above)
Comment (jasperswart) [reply src/signals/incident-management/containers/IncidentDetail/components/Header/components/DownloadButton/index.js]: doen we later

**Pull request merged**